### PR TITLE
added maven pom file, moved files into maven dir structure, bitcoinj 13

### PR DESCRIPTION
**Pull request by mrosseel**
my project is based on maven and didn't want to copy/paste the bitpay code in for maintainability reasons, so I mavenized it so it can be 'mvn install'ed locally and my project can depend on it.
Willing to fix up any issues you guys see, I wrote something similar for the bitpay v1 api, hope this one gets merged :)

### CODE REVIEW

File: src/main/java/controller/BitPay.java [hunk 1]
```diff
@@ -33,13 +33,14 @@
...
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.bitcoin.core.ECKey;
+import org.bitcoinj.core.ECKey;
 
 public class BitPay {
 
     private static final String BITPAY_API_VERSION = "2.0.0";
     private static final String BITPAY_PLUGIN_INFO = "BitPay Java Client " + BITPAY_API_VERSION;
-    private static final String BITPAY_URL = "https://bitpay.com/";
+    public static final String BITPAY_URL = "https://bitpay.com/";
```
Comment (kleetus) [inline src/main/java/controller/BitPay.java]: why turn this into a public string? I don't see this change being leveraged anywhere? It is a super small thing, but maybe I am missing something?

**Comment by kleetus** [comment]
I will handle the merge conflicts with this branch. It was my fault that they occurred anyway. This merge request is badly needed and greatly appreciated.

File: src/main/java/controller/KeyUtils.java [hunk 2]
```diff
@@ -8,23 +8,34 @@
...
-    
+    final private static String PRIV_KEY_FILENAME = "/bitpay_private.key";
+
     public KeyUtils() {}
-    
+
     public static boolean privateKeyExists()
     {
-        return new File(PRIV_KEY_FILENAME).exists();
+        return getPrivateKeyFile().exists();
```
Comment (kleetus) [inline src/main/java/controller/KeyUtils.java]: Small thing here, this is problematic if PRIV_KEY_FILENAME is not a resource. There would be an uncaught NPE. I think this is a good way to handle this (using resources), but we might want to set this up properly.

**Comment by kleetus** [comment]
@ajp8164 I think I am going to merge this in and make one or two fixes on top. This is mainly moving the project to maven. This is a good thing overall.

**Comment by kleetus** [comment]
I've added most of this merge request into #17, closing this one. @mrosseel you commits will still be merged later on.

File: src/main/java/controller/BitPay.java (hunk 1 above)
Comment (mrosseel) [inline src/main/java/controller/BitPay.java]: I'm using the bitpay client as a dependency in my software where I use this string. Not a big deal, but one of the use cases of this library is to perform bitpay stuff on the server and present the invoice in the web client. For that, the invoice string needs to be constructed, and for that I need the bitpay url. I also added whether we're in test/production. No prob if you leave this out, I agree it's a quick hack, but support for building the URL should be in the library imho.

File: src/main/java/controller/KeyUtils.java (hunk 2 above)
Comment (mrosseel) [inline src/main/java/controller/KeyUtils.java]: yeah, needed this in production - I'm running in jetty so reading from classpath is better than absolute paths which differ between dev/test/prod. Agreed on the NPE, let me know if you want me to fix it, not sure what you did in the other pull request.
Note: my maven pull request of the version 1 bitpay client was never merged so I did not think anything would happen with this one, thanks for picking it up ;)

**Pull request closed without merging**